### DISCUSSION TITLE
IGMPproxy failed MRT_INIT failed; Errno(92): Protocol not available

### DISCUSSION THREAD

**Discussion by bclevering** [fabianishere/udm-iptv#132] [Q&A]
I've got a problem to get IPTV (KPN) to work.

The big problem is that i got a new device (UDW (Unifi Dream Wall)) in early access... (looks very nice)

i found out that the ip address (WAN ethernet port is eth18 or should i use ppp0?) but it's still not working:

Please share the following output with the developers:
=== Configuration ===
WAN Interface: eth18
WAN VLAN: 4 (dev iptv)
WAN DHCP: true (options "-O staticroutes -V IPTV_RG")
WAN Ranges: 213.75.0.0/16 217.166.0.0/16
LAN Interfaces: br0
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: true
=== IP Link and Route ===
=== Service Logs ===
Jun 02 13:05:01 Dream-Wall udm-iptv[12418]: Sending discover...
Jun 02 13:05:01 Dream-Wall udm-iptv[12418]: Sending select for 10.220.101.245...
Jun 02 13:05:01 Dream-Wall udm-iptv[12418]: Lease of 10.220.101.245 obtained, lease time 3600
Jun 02 13:05:01 Dream-Wall udm-iptv[12418]: NATing IPTV network ranges (if necessary)
Jun 02 13:05:01 Dream-Wall udm-iptv[12418]: Setting up igmpproxy
Jun 02 13:05:01 Dream-Wall udm-iptv[12418]: Starting igmpproxy
Jun 02 13:05:01 Dream-Wall udm-iptv[12418]: MRT_INIT failed; Errno(92): Protocol not available
Jun 02 13:05:01 Dream-Wall systemd[1]: udm-iptv.service: Main process exited, code=exited, status=255/EXCEPTION
Jun 02 13:05:01 Dream-Wall systemd[1]: udm-iptv.service: Failed with result 'exit-code'.
Jun 02 13:05:01 Dream-Wall systemd[1]: udm-iptv.service: Consumed 166ms CPU time.

======

ip -4 addr show
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
15: br0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    inet 192.168.2.254/22 scope global br0
       valid_lft forever preferred_lft forever
56: ppp0: <POINTOPOINT,MULTICAST,NOARP,UP,LOWER_UP> mtu 1492 qdisc pfifo_fast state UNKNOWN group default qlen 3
    inet <IP NUMBER FROM KPN> peer 195.190.228.51/32 scope global ppp0
       valid_lft forever preferred_lft forever

=====

192.168.0.0/22 dev br0 proto kernel scope link src 192.168.2.254 
195.190.228.51 dev ppp0 proto kernel scope link src <IP NUMBER FROM KPN>

**Comment by fabianishere** [comment]
Based on the following log:
>  MRT_INIT failed; Errno(92): Protocol not available

It looks like the Linux kernel on the UniFi Dream Wall does not have multicast routing enabled. You'll have to ask Ubiquiti to enable this feature for newer firmware releases or try to port [udm-kernel-tools](https://github.com/fabianishere/udm-kernel-tools) to that platform.

**Reply by bclevering** [reply]
Thanks for your quick response! it looks like i cannot port the udm-kernel-tools because of some issues with the build tools:

root@Dream-Wall:~# apt install build-dep libncurses-dev gawk flex bison openssl libssl-dev gcc-aarch64-linux-gnu
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Note, selecting 'libncurses5-dev' instead of 'libncurses-dev'
Package gcc-aarch64-linux-gnu is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Unable to locate package build-dep
E: Package 'gcc-aarch64-linux-gnu' has no installation candidate


Is this manual still actual or should i use some other tooling? Thanks in advance!

**Reply by fabianishere** [reply]
It uses a cross-compiler, so you should build it from your development machine.

**Reply by bclevering** [reply]
sounds like a lot of work? I could try this, but i think i need some help setting things up and letting it work for the Dream Wall...

Can someone help?

**Reply by bclevering** [reply]
@fabianishere i tried compiling the kernel but it keeps saying that ssh-proxy cannot be found...

Looks like it's an issue because "unifi-os shell" also isn't working because i go straight into the shell...

**Reply by fabianishere** [reply]
The firmware is probably similar to the UDM SE where there is not a container anymore and the controller runs directly on the root filesystem.

**Reply by bclevering** [reply]
@fabianishere looks like it indeed;

**Comment by bclevering** [comment]
Should I try this solution I found on tweakers.net? https://github.com/mories76/udmprose-iptv

**Reply by fabianishere** [reply]
Thats based on an old version of this project and also requires kernel support for multicast routing, so it won’t work probably.
